ClothesApp/Application/CategoryTree.cs

```diff
@@ -0,0 +1,57 @@
+namespace Application;
+
+public static class CategoryTree
+{
+    public interface ITree<T>
+    {
+        T Data { get; }
```

Comment:
This interface and model doesn't make sense for client
Why would you return leaf root and all this stuff to client?
It should just be Category { int Id, string Name, List<Category> Subcategories }